File: Data Structures Lab using Python/2. Searching/binarySearch.py
#using binary search concept and taking input seperately
#we can also use .split method to take input
size=int(input("Enter size of list: "))
store=[]
for i in range(size):
	element=int(input("Enter your number: "))
	store.append(element)
store.sort()
print("After sorting list is: ",store)
length=len(store)
low=0
high=length-1
value=int(input("The number to search for: "))
loc=-1
while(low<=high):
	mid=(low+high)//2
	if value==store[mid]:
		loc=mid+1
		break
	elif value>store[mid]:
		low=mid+1
	elif value<store[mid]:
		high=mid-1
		low=0
if loc!=-1:
	print(value,"was found at index",str(loc-1)+'.')
	
	

File: Design and Analysis of Algorithm Lab/1. Searching & Sorting/binarySearch.c
#include<stdio.h>
int binary(int arr[],int k,int low,int high)
{
	if(low<=high)
	{
		int mid=(low+high)/2;
		if(arr[mid]==k)
		{
			return mid;
		}
		else if(arr[mid]<k)
		{
			return binary(arr,k, mid+1,high);
		}
		return binary(arr,k,low,mid-1);
	}
	return-1;
}
int main()
{
	int n,k;
	printf("Enter number of elements: ");
	scanf("%d",&n);
	printf("Enter the sorted array: ");
	int arr[n];
	for(int i=0;i<n;i++)
	{
		scanf("%d",&arr[i]);
	}
	printf("enter the item to be search: ");
	scanf("%d",&k);
	int a=binary(arr,k,0,n-1);
	if(a==-1)
	{
		printf("item not present");
	}
	else
	{
		printf("item present");
	}
	return 0;
}
